static/panes/executor.js

```diff
@@ -481,7 +483,7 @@ Executor.prototype.initListeners = function () {
     this.container.on('resize', this.resize, this);
     this.container.on('shown', this.resize, this);
     this.container.on('open', function () {
-        this.eventHub.emit('compilerOpen', this.id, this.sourceEditorId);
+        this.eventHub.emit('executorOpen', this.id, this.sourceEditorId);
```

Comment:
I _think_ this (or something similar in these message changes) have introduced a bug. When I was testing locally, opening a new executor pane no longer gets the initial text of the editor on startup.

To repro:

* reset UI
* create a simple program (`#include <cstdio> int main() {puts("hello world\n"); }`)
* Check it compiles ok
* Create a new execution-only view and note that its contents are empty (no compile happened)
* Change the compile flags (e.g. to `-O1`) on the exec view and note the error is "linker no main"
* Now modify the source and see alls' ok

I _think_ these symptoms point to the editor not sending text any more. I think there's a spot in the editor that responds to "compilerOpen" or similar that re-sends its text, triggering a build. And that might now need to look at executroOpen too.

Reply:
Should be fixed, I think it only worked because this.id used to be undefined. Added the executorOpen eventhandler to reflect that in de editor.

Reply:
Thanks for digging deeper and finding the actual cause of it working before :blush: :) Nice one!